ambry-router/src/main/java/com/github/ambry/router/PutOperation.java

```diff
@@ -120,6 +120,14 @@ class PutOperation {
   private boolean isEncryptionEnabled;
   private final QuotaChargeCallback quotaChargeCallback;
 
+  // Whether the entire blob (all chunks) are compressible.  Default is null for undecided.
+  // The value is set based on blobProperties and compression service configuration.
+  // For example, if BlobProperties.ContentEncoding specified, then the entire blob will
+  // not be compressed by setting this field to FALSE.
+  // This field is set once when inserting the first chunk based on blob properties.
+  // All chunk compression will check this field and skip compression if this value FALSE.
+  private Boolean isBlobCompressible;
```

Comment:
could just assign the value to this variable in the constructor, then you don't need to check if it's null for each chunk.

Reply:
I agree.  Let me make that change.